providers/curation/github.js

```diff
@@ -139,11 +140,42 @@ class GitHubCurationService {
       fileBody.committer = { name: info.name || info.login, email: info.email }
     if (get(currentContent, '_origin.sha')) {
       fileBody.sha = currentContent._origin.sha
+
+      try {
+        let validateResult = await this._validateComponentDefinitionExists(patch)
+        for (var elem of validateResult) {
+          if (elem.revision) {
+            return response.status(400).send({ errors: `component does not exist: ${elem.revision.toString()}` })
```

Comment:
where is `response` coming from? This is the service that could be called from anywhere. response and status codes should stay in the route.

Reply:
Related, perhaps validation should go in `addOrUpdate`. There we can just use `definitionService.getAll()` having created a complete list of all the definitions affected in the curation. If some of them are missing then we can report back the missing list and bail on the entire operation.

Reply:
from analysis it appears that if we use definitionService.getAll() , it will fail the entire PATCH if only one of the definition/revision is missing. Because of this behaviour we have implemented a solution that uses definitionService.get(coordinates) instead